packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -202,3 +203,28 @@ export const typeMatchesSomeSpecifier = (
   program: ts.Program,
 ): boolean =>
   specifiers.some(specifier => typeMatchesSpecifier(type, specifier, program));
+
+export function valueMatchesSpecifier(
+  node: TSESTree.Node,
+  specifier: TypeOrValueSpecifier,
+): boolean {
+  if ('name' in node && typeof node.name === 'string') {
+    if (typeof specifier === 'string') {
+      return node.name === specifier;
+    }
+
+    if (typeof specifier.name === 'string') {
+      return node.name === specifier.name;
+    }
+
+    return specifier.name.includes(node.name);
```

Comment:
[Bug] This just checks the specifier name. But it doesn't check where the specifier comes from. This'll need to more closely mirror how `typeMatchesSpecifier` switches on `specifier.from`. Instead of just checking the `specifier.name`, the place the specifier is declared needs to be checked too.

You might need to use the scope manager for this, to see where the value actually comes from locally.